BraintreeCore/src/main/java/com/braintreepayments/api/DeviceInspector.kt

```diff
@@ -0,0 +1,154 @@
+        return appHelper.isAppInstalled(context, VENMO_APP_PACKAGE)
+    }
+
+    private val isDeviceEmulator: Boolean
+        get() = "google_sdk".equals(Build.PRODUCT, ignoreCase = true) ||
+                "sdk".equals(Build.PRODUCT, ignoreCase = true) ||
+                "Genymotion".equals(Build.MANUFACTURER, ignoreCase = true) ||
+                Build.FINGERPRINT.contains("generic")
+
+    private fun getAppName(context: Context?): String =
+        getApplicationInfo(context)?.let { appInfo ->
+                context?.packageManager?.getApplicationLabel(appInfo).toString()
+        } ?: "ApplicationNameUnknown"
+
+    private fun getApplicationInfo(context: Context?) =
+        try {
+            context?.packageManager?.getApplicationInfo(context.packageName, 0)
+        } catch (e: PackageManager.NameNotFoundException) {
+            null
+        }
+
+    private fun getNetworkType(context: Context?): String =
+        context?.let {
+            val connectivityManager =
+                it.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+            connectivityManager.activeNetworkInfo?.typeName
+        }?: "none"
+
+    private fun getAppVersion(context: Context?): String = getPackageInfo(context) ?: "VersionUnknown"
+
+    private fun getPackageInfo(context: Context?) =
+        context?.let {
+            try {
+                val packageInfo = it.packageManager.getPackageInfo(it.packageName, 0)
+                packageInfo?.versionName
+            } catch (ignored: PackageManager.NameNotFoundException) { null }
+        }
+
+    private fun getUserOrientation(context: Context?): String =
+        when (context?.resources?.configuration?.orientation ?: Configuration.ORIENTATION_UNDEFINED) {
+            Configuration.ORIENTATION_PORTRAIT -> "Portrait"
+            Configuration.ORIENTATION_LANDSCAPE -> "Landscape"
+            else -> "Unknown"
+        }
+
+    /**
+     * Gets the current Drop-in version or null.
+     *
+     * @return string representation of the current Drop-in version, or null if
+     * Drop-in is unavailable
+     */
+    private val dropInVersion
+        get() = getDropInVersion()
+
+    companion object {
+        private const val PAYPAL_APP_PACKAGE = "com.paypal.android.p2pmobile"
+        private const val VENMO_APP_PACKAGE = "com.venmo"
+        private const val VENMO_APP_SWITCH_ACTIVITY = "controller.SetupMerchantActivity"
+
+        @VisibleForTesting
+        const val VENMO_BASE_64_ENCODED_SIGNATURE = "x34mMawEUcCG8l95riWCOK+kAJYejVmdt44l6tzcyUc=\n"
+        private val venmoIntent: Intent
+            get() = Intent().setComponent(
+                ComponentName(
+                    VENMO_APP_PACKAGE,
+                    "$VENMO_APP_PACKAGE.$VENMO_APP_SWITCH_ACTIVITY"
+                )
+            )
+
+        internal fun getDropInVersion(): String? {
+            try {
+                val clazz = Class.forName("com.braintreepayments.api.dropin.BuildConfig")
```

Comment:
```suggestion
                val class = Class.forName("com.braintreepayments.api.dropin.BuildConfig")
```
^ assuming this was a typo?

Reply:
It's legit `class` is a reserved word in Java and Kotlin.

Reply:
:actual-lol: maybe we can do `val dropInClass` or something instead then? or is it any form? I guess `class` is also protected for properties in swift by itself unless you escape it.